SimG4CMS/Tracker/interface/TkAccumulatingSensitiveDetector.h

```diff
@@ -46,56 +45,53 @@ public Observer<const BeginOfJob*>
     void fillHits(edm::PSimHitContainer&, const std::string&) override;
     void clearHits() override;
 
-    std::string type();
-
 private:
-    virtual void sendHit();
-    virtual void updateHit(G4Step *);
-    virtual bool newHit(G4Step *);
-    virtual bool closeHit(G4Step *);
-    virtual void createHit(G4Step *);
-    void checkExitPoint(Local3DPoint);
+    void createHit(const G4Step *);
+    void sendHit();
+    void updateHit(const G4Step *);
+    bool newHit   (const G4Step *);
+    bool closeHit (const G4Step *);
+
+    void checkExitPoint(const Local3DPoint&);
     void update(const BeginOfEvent *) override;
     void update(const BeginOfTrack *) override;
     void update(const BeginOfJob *) override;
-    Local3DPoint toOrcaRef(Local3DPoint ,G4VPhysicalVolume *);
-    int tofBin(float);
 
+    Local3DPoint toOrcaRef(const Local3DPoint&);
+    TrackInformation* getTrackInformation(const G4Track *);
+
+    // data members initialised before run
     TrackingSlaveSD * slaveLowTof;
     TrackingSlaveSD * slaveHighTof;
     FrameRotation * myRotation;
-    UpdatablePSimHit * mySimHit;
-    std::string pname;
-    Local3DPoint globalEntryPoint;
-    Local3DPoint globalExitPoint;
     const SimTrackManager* theManager;
-    G4VPhysicalVolume * oldVolume;
-    G4ProcessTypeEnumerator * theG4ProcessTypeEnumerator;
-    double theSigma;
-    uint32_t lastId;
-    unsigned int lastTrack;
-    int eventno;
-    // cache stuff for debugging
-    float px,py,pz;
+    const G4ProcessTypeEnumerator * theG4ProcessTypeEnumerator;
+    const G4TrackToParticleID * myG4TrackToParticleID;
+    TrackerG4SimHitNumberingScheme* numberingScheme_;
     bool allowZeroEnergyLoss;
     bool printHits;
     bool neverAccumulate;
-    G4TrackToParticleID * myG4TrackToParticleID;
-    TrackInformation* getOrCreateTrackInformation(const G4Track *);
+    double rTracker2;   // tracker volume R^2
+    double rTracker;    // tracker volume R
```

Comment:
were float, now double. why?

Reply:
it is compared with double, not part of hit, so saving of memory is not important.